xonsh/base_shell.py

```diff
@@ -343,11 +343,14 @@ def _append_history(self, tee_out=None, **info):
         information is available.
         """
         hist = builtins.__xonsh_history__  # pylint: disable=no-member
+        env = builtins.__xonsh_env__
         info['rtn'] = hist.last_cmd_rtn
         tee_out = tee_out or None
         last_out = hist.last_cmd_out or None
         if last_out is None and tee_out is None:
             pass
+        elif not env.get('XONSH_STORE_STDOUT'):
+            pass
```

Comment:
Clarifying question: we want `__xonsh_history__.outs` to remain accessible but only write those outs to file if `XONSH_STORE_STDOUT` is `True`?